administrator/components/com_media/src/Event/FetchMediaFilesEvent.php

```diff
@@ -0,0 +1,196 @@
+		if (!\array_key_exists('files', $arguments) || !is_array($arguments['files']))
+		{
+			throw new BadMethodCallException("Argument 'files' of event $name is not of the expected type");
+		}
+	}
+
+	/**
+	 * Validate $value to be an array
+	 *
+	 * @param   array  $values  The value to set
+	 *
+	 * @return array
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function setFiles(array $values): array
+	{
+		$result = [];
+
+		foreach($values as $value) {
+			$result[] = $this->validateFile($value);
+		}
+
+		return $result;
+	}
+
+	protected function getFiles(array $values) {
+		$result = [];
+
+		foreach($values as $value) {
+			$result[] = clone $value;
+		}
+
+		return $result;
+	}
+
+		/**
+		 * Validate $value to have all attributes with a valid type
+		 *
+		 * Validation based on \Joomla\Component\Media\Administrator\Adapter\AdapterInterface::getFile()
+		 *
+		 * Properties validated:
+		 * - type:          The type can be file or dir
+		 * - name:          The name of the file
+		 * - path:          The relative path to the root
+		 * - extension:     The file extension
+		 * - size:          The size of the file
+		 * - create_date:   The date created
+		 * - modified_date: The date modified
+		 * - mime_type:     The mime type
+		 * - width:         The width, when available
+		 * - height:        The height, when available
+		 *
+		 * Generation based on \Joomla\Plugin\Filesystem\Local\Adapter\LocalAdapter::getPathInformation()
+		 *
+		 * Properties generated:
+		 * - created_date_formatted:  DATE_FORMAT_LC5 formatted string based on create_date
+		 * - modified_date_formatted: DATE_FORMAT_LC5 formatted string based on modified_date
+		 *
+		 * @param   \stdClass  $value  The value to set
+		 *
+		 * @return \stdClass
+		 *
+		 * @since   __DEPLOY_VERSION__
+		 *
+		 * @throws BadMethodCallException
+		 */
+		private function validateFile(\stdClass $value): \stdClass
+		{
+				// Make immutable object
+				$value = clone $value;
+
+				// Only "dir" or "file" is allowed
+				if (!isset($value->type) || ($value->type !== 'dir' && $value->type !== 'file'))
+				{
+						throw new BadMethodCallException("Property 'type' of argument 'file' of event {$this->name} has a wrong value. Valid: 'dir' or 'file'");
+				}
+
+				// Non empty string
+				if (empty($value->name) || !is_string($value->name))
+				{
+						throw new BadMethodCallException("Property 'name' of argument 'file' of event {$this->name} has a wrong value. Valid: non empty string");
+				}
+
+				// Non empty string
+				if (empty($value->path) || !is_string($value->path))
+				{
+						throw new BadMethodCallException("Property 'path' of argument 'file' of event {$this->name} has a wrong value. Valid: non empty string");
+				}
+
+				// A string
+				if (!isset($value->extension) || !is_string($value->extension))
+				{
+						throw new BadMethodCallException("Property 'extension' of argument 'file' of event {$this->name} has a wrong value. Valid: string");
+				}
```

Comment:
```suggestion
		if ($value->type === 'file'))
		{
				// Non empty string
				if (empty($value->path) || !is_string($value->path))
				{
						throw new BadMethodCallException("Property 'path' of argument 'file' of event {$this->name} has a wrong value. Valid: non empty string");
				}

				// A string
				if (!isset($value->extension) || !is_string($value->extension))
				{
						throw new BadMethodCallException("Property 'extension' of argument 'file' of event {$this->name} has a wrong value. Valid: string");
				}
		}
```

hopefully it's correct